File: forest/hold-for-reinforcements.lua
-- using Long Sword, Steel Striker
loop
    local enemy = self:findNearest(self:findEnemies())
    local flag = self:findFlag()
    if flag then
        self:pickUpFlag(flag)
    elseif enemy then
        if self:isReady("cleave") then
            self:cleave(enemy)
        elseif self:isReady("bash") then
            self:bash(enemy)
        else
            self:attack(enemy)
        end
    end
end

File: forest/hold-for-reinforcements.clj
(while true
  (def f (.findFlag this))
  (if f
    (.pickUpFlag this f)
    (when true
      (def e (.findNearestEnemy this))
      (if e
        (if (.isReady this "bash")
          (.bash this e)
          (.attack this e))
        (.wait this 0.1)))))
